common/src/main/java/earth/terrarium/olympus/client/components/Widgets.java

```diff
@@ -38,4 +44,13 @@ public static Button toggle(State<Boolean> state) {
         return toggle(state, button -> {
         });
     }
+
+    public static <T> Button dropdown(DropdownState<T> state, List<T> options, Function<T, Component> optionText, Consumer<Button> factory, Consumer<DropdownBuilder<T>> builder) {
```

Comment:
have it the other way around and have the factory be applied after the dropdown init so that the dropdown renderer can be edited if needed